go-controller/pkg/metrics/ovn_db.go

```diff
@@ -351,7 +351,7 @@ func getOvnDbVersionInfo() {
 
 func RegisterOvnDBMetrics(clientset kubernetes.Interface, k8sNodeName string) {
 	err := wait.PollImmediate(1*time.Second, 300*time.Second, func() (bool, error) {
-		return checkPodRunsOnGivenNode(clientset, "name=ovnkube-db", k8sNodeName, false)
+		return checkPodRunsOnGivenNode(clientset, "ovn-db-pod=true", k8sNodeName, false)
```

Comment:
@girishmg will this affect you for upgrade?

Reply:
@trozet sorry missed this message.  Thanks for checking. 

@abhat can you instead use this? 

"name in (ovnkube-db, whatever_openshift_uses_for_name)"

For example:

```
kubectl get pods -n ovn-kubernetes -l 'name in (ovn-north, onvkube-db)'
NAME                        READY   STATUS    RESTARTS   AGE
ovn-north-77b9747cd-kwf4h   1/1     Running   0          33m
ovn-north-77b9747cd-m9vds   1/1     Running   0          46m
ovn-north-77b9747cd-pmxv5   1/1     Running   0          46m
```

Reply:
@girishmg we use `name=ovnkube-master` for the pods that run the db. will that conflict with the ovnkube-master pods that you run so for instance, this will change to `name in (ovnkube-db, ovnkube-master)`. Will it select the master pods on some nodes for db metrics when it shouldn't?

Reply:
Right, that will be an issue.

However, Shouldn’t this change be in downstream `openshift` only? Your PR also doesn’t include labelling the ovnkube-db.yaml and ovnkube-db-raft.yaml,  so if someone uses the upstream yamls with your changes to ovn-db-pod=true , then it will not work for them.

Reply:
The upstream templates are already updated to have `ovn-db-pod=true` in #1747

Reply:
But we figured that actually breaks upgrade for us downstream because the matchLabels in the selector is immutable. So it is likely that my change to the templates in #1747 may break your upgrade case as well. I am working on the PR to revert that and add a more appropriate check.

Reply:
It turns out that the upstream doesn't add the `ovn-db-pod=true` to `.spec.selector.matchLabels` of the daemonset but only to the `.metadata.labels` of the pod. So the above issue I mentioned is a downstream only problem introduced in CNO.